colcon-test/action.yaml

```diff
@@ -0,0 +1,119 @@
+name: colcon-test
+description: ""
+
+inputs:
+  rosdistro:
+    description: ""
+    required: true
+  target-packages:
+    description: ""
+    required: true
+  build-depends-repos:
+    description: ""
+    required: false
+  label:
+    description: ""
+    required: false
+    default: ".*" # Run all tests
+  token:
+    description: ""
+    required: false
+    default: ${{ github.token }}
+outputs:
+  coverage-reports:
+    description: ""
+    value: ${{ steps.test.outputs.coverage-reports }}
+
+runs:
+  using: composite
+  steps:
+    - name: Show target packages
+      run: |
+        echo "target packages: ${{ inputs.target-packages }}"
+      shell: bash
+
+    - name: Set up colcon-mixin
+      run: |
+        colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml || true
+        colcon mixin update default
+      shell: bash
+
+    - name: Install packages for coverage commands
+      run: |
+        sudo apt-get -yqq update
+        sudo apt-get -yqq install lcov python3-pip
+        pip3 install colcon-lcov-result colcon-coveragepy-result
+      shell: bash
+
+    - name: Clone dependency packages
+      if: ${{ inputs.build-depends-repos != ''}}
+      run: |
+        git config --local --unset-all http.https://github.com/.extraheader || true
+        git config --global url.https://x-access-token:${{ inputs.token }}@github.com.insteadOf 'https://github.com'
+        mkdir dependency_ws
+        vcs import dependency_ws < ${{ inputs.build-depends-repos }}
+      shell: bash
+
+    - name: Run rosdep install
+      run: |
+        sudo apt-get -yqq update
+        rosdep update
+        DEBIAN_FRONTEND=noninteractive rosdep install -yqq --from-paths . --ignore-src --rosdistro ${{ inputs.rosdistro }}
+      shell: bash
+
+    - name: Restore build files from cache
+      id: restore-build-files
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./build
+          ./install
+        key: build-${{ github.sha }}
+
+    - name: Test
+      id: test
+      run: |
+        set -e
+        . /opt/ros/${{ inputs.rosdistro }}/setup.sh
+
+        colcon lcov-result --zero-counters \
```

Comment:
[Question] What's `--zero-counters`? What's the difference between `--initial`?

Reply:
Also, is there no initialization for `coveragepy-result`?

Reply:
> [Question] What's `--zero-counters`? What's the difference between `--initial`?

A zero-counters resets the coverage report file. And initial option creates a coverage report file if it doesn't exist.
https://github.com/colcon/colcon-lcov-result#usage

Reply:
> Also, is there no initialization for `coveragepy-result`?

For coveragepy, I don't think the initialization step is needed

Reply:
Thank you. I understand that it's a more polite way.